lib/ansible/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -167,13 +184,25 @@ def native_python_main(this_gitlab, purge, var_list, state):
     existing_variables = [x.get_id() for x in gitlab_keys]
 
     for key in var_list:
+
+        if isinstance(var_list[key], str):
```

Comment:
```suggestion
        if isinstance(var_list[key], string_types):
```
For that you need `from ansible.module_utils.six import string_types`.

Reply:
What happens if `var_list[key]` happens to be an integer (or a float, or a boolean, or a list)? Now it will be treated as a `dict` (and the module will die). What did happen in the previous version? Did `python-gitlab` complain? Or was it converted to a string?

Reply:
Basically gitlab project variables are available as environment variables inside gitlac ci jobs. And basically an environment variable are always strings (afaik).  
But also here, I have to research.

Reply:
Values can be strings, integers or floats. python-gitlab accept both.  
lists and dictionary results in `gitlab.exceptions.GitlabCreateError: 400: value is invalid`

so maybe a value should be validated